### DISCUSSION TITLE
Google fit

### DISCUSSION THREAD

**Discussion by Gdii** [indomus/forum#1976] [HUB: Home Assistant]
Salve,
premesso di essere entusiasta ma incompetente riguardo HA, chiedo aiuto per l'integrazione del componente Google fit  in Home Assistant-OS.
seguendo la procedura  trovata qui:     https://github.com/IvanVojtko/googlefit-homeassistant   sono riuscito a crearmi le credenziali Google, ma ora sono bloccato alla fase di configurazione su HA.
nel senso che non ho idea di come realizzare questo:

"To allow HA access your Fit data, you need to complete a challenge. It can't be completed by HA so that's why you need to use get_credentials.py script. First install all the requirements using python -m pip install -r requirements.txt. Then run script python get_credentials.py, open the generated URL, allow access and don't forget to tick mark all permissions. This script will generate .google_fit.token file. Copy this file to your HA configuration directory "

...qualche anima buona ha voglia di darmi una mano ?!

grazie mille, 
Gdi.

**Comment by Gdii** [comment]
Salve,
proprio nessuno puo' aiutarmi ???
grazie !

**Reply by snakuzzo** [reply]
Devi aprire una shell (magari, se non l'hai fatto, installa l'addon "SSH & Web Terminal". Apri una shell...
vai sotto la dir del custom component e fai quello che è indicato nelle istruzioni

**Reply by Gdii** [reply]
ok, avevo già installato "SSH & Web Terminal"...però mi sfugge nel dettaglio cosa devo fare....avrei bisogno di un'assistenza...."step by step"
capisco di essere un pò in alto mare, ma non ho idea di come potermi arrangiare altrimenti.
grazie anticipate per l'eventuale disponibilità.

**Reply by snakuzzo** [reply]
Ma te l'ho già scritto...ed è tutto scritto pure nelle istruzioni sul progetto.
Comunque, da quello che ho capito sei arrivato al punto in cui hai sul filesystem questa directory...

```
/config/custom_components/google_fit
```

...e sempre sotto questa directory avrai il file `credentials.json`

A questo punto apri il terminale e digiti questo...

```
cd /config/custom_components/google_fit
python3 -m pip install -r requirements.txt
python3 get_credentials.py
```

Verrà generata una URL. La copi e la incolli in un browser (dove sarai sicuramente già loggato su google), concedi i permessi e lo script genererà il file `.google_fit.token`. Prendi questo file e lo sposti sotto la directory `/config/`.

Immagino che poi sarà necessario riavviare HA.

Fine

**Reply by Gdii** [reply]
ok, perfetto...in effetti fino alla directory google_fit ci arrivo....ma non so come installare python, infatti, al comando successivo mi restituisce
"zsh: command not found: python"

**Reply by snakuzzo** [reply]
Non devi installare python....python già c'è...solo che non c'è il symlink `python`, quindi devi usare `python3` (correggo nell'altro post)

![image](https://user-images.githubusercontent.com/7205371/196632610-327956d1-7c1d-4138-be73-5df92cdef3fe.png)

**Reply by Gdii** [reply]
è quasi avvvenuto il miracolo.....
nel senso che tutto ha funzionato "lato HA"....ha generato il link, mi sono collegato, nonostante mi abbia dato il warning dell'applicazione non verificata da Google....però ho continuato, dato il consenso all'accesso dei dati, ma quando si è trattato di confermare....la risposta è stata "connessione negata da localhost "ERR_CONNECTION_REFUSED"....
eppure lato OAuth mi pare funzioni tutto...

**Reply by snakuzzo** [reply]
Chi non si connette a cosa?
Controlla i log di Core quando provi a completare...non guardare solo quello che ti appare sulla webgui (sempre se parliamo di un errore lato HA)

**Reply by Gdii** [reply]
no, appunto, lato HA, direi che va tutto...mi apre la pagina di google in cui devo collegarmi al mio account e autorizzare il trattamento dei dati...ma quando concludo con "conferma", mi scrive quello

**Reply by Gdii** [reply]
![cat2](https://user-images.githubusercontent.com/17603858/196781399-774a7fd1-d18b-4e1f-8522-f271da655488.png)
![cat3](https://user-images.githubusercontent.com/17603858/196781402-52a18727-9908-4f9a-b797-aa3b09ddf31e.png)

**Reply by snakuzzo** [reply]
Ma sul browser (barra indirizzi) cosa prova a raggiungere?

**Reply by Gdii** [reply]
http://localhost:8080/?state=Hk6wggXD4Bdrb1t02jWQH87rRXTX31&code=4/0ARtbsJqcWj01i88fDNjALucB94RfoRWvpbrUvK_fGFxfz4f9CeP-51RSAFf2aNGrwfUHBQ&scope=https://www.googleapis.com/auth/fitness.sleep.read%20https://www.googleapis.com/auth/fitness.heart_rate.read%20https://www.googleapis.com/auth/fitness.oxygen_saturation.read%20https://www.googleapis.com/auth/fitness.blood_pressure.read%20https://www.googleapis.com/auth/fitness.nutrition.read%20https://www.googleapis.com/auth/fitness.body.read%20https://www.googleapis.com/auth/fitness.location.read%20https://www.googleapis.com/auth/fitness.activity.read

**Reply by snakuzzo** [reply]
C'è qualcosa di errato. Quel puntamento a localhost sulla 8080 non ha senso. Rileggi bene la documentazione del custom

**Reply by Gdii** [reply]
ma è che la fa in automatico....nel senso che da quella pagina di configurazione delle autorizzazioni, premendo "continua"....punta qui...e io non è che ci metto del mio.....

**Reply by snakuzzo** [reply]
Quando ho tempo do un'occhiata e lo provo...ma non so dirti quando

**Reply by Gdii** [reply]
ok, grazie mille, sei stato molto gentile...intanto io provo a guardare in giro, anche se la vedo un filo disperata.
Buona serata !!!

**Reply by Gdii** [reply]
Ciao, ho continuato a guardare e fare le probe più disparate....ma nulla...non è che per caso hai un'idea ?!

**Reply by Gdii** [reply]
Buonasera,
ho ripreso in mano questa integrazione, reinstallandola su una nuova istanza di Home Assistant OS....purtroppo con lo stesso risultato...
non è che per caso, ti è riuscito di guardarci ?!

Grazie anticipate
Gdi.
